MetaMask/metamask-extension
# Hide 'Add snap account' button behind an experimental setting

## **Description**

In preparation for the stable release of the Snap Accounts API, we would like to hide the "Add Snap account" button from the accounts list, placing it behind an experimental setting.

To achieve this, this PR adds a new section in the experimental settings with a toggle to enable or disable the "Add Snap account" button. It defaults to disabled.

We also track a new event called `Add Snap Account Enabled`.

## **Manual testing steps**

1. Install Flask and onboard your SRP.
2. Go to your accounts list.
3. The "Add Snap account" must not be present.
4. Go to "Settings" > "Experimental".
5. Enable the "Add Snap account" button.
6. Go to your accounts list.
7. The "Add Snap account" must be present.
8. Click on the "Add Snap account"
9. You should see the list of allowed Account Snaps.

## **Screenshots/Recordings**

### New experimental settings:

<img width="318" alt="image" src="https://github.com/MetaMask/metamask-extension/assets/68558152/bc3e43c5-349b-4ffb-b835-3b6ddfbdf525">

### Accounts list when "Add Snap account" is disabled:

<img width="340" alt="image" src="https://github.com/MetaMask/metamask-extension/assets/68558152/1d34af2e-da26-4914-be41-dff124be1889">

### Accounts list when "Add Snap account" is enabled:

<img width="337" alt="image" src="https://github.com/MetaMask/metamask-extension/assets/68558152/37f8913e-3762-4790-893a-941fa39831fd">

## **Related issues**

Fixes: MetaMask/accounts-planning#45

## **Pre-merge author checklist**


## **Pre-merge reviewer checklist**
